sdk/core/azure-core/src/main/java/com/azure/android/core/http/interceptors/AddDateInterceptor.java

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.android.core.http.interceptors;
+
+import androidx.annotation.NonNull;
+
+import org.threeten.bp.OffsetDateTime;
+import org.threeten.bp.ZoneId;
+import org.threeten.bp.format.DateTimeFormatter;
+
+import java.io.IOException;
+import java.util.Locale;
+
+import okhttp3.Interceptor;
+import okhttp3.Response;
+
+/**
+ * The pipeline interceptor that adds a "Date" header in RFC 1123 format when sending an HTTP request.
+ */
+public class AddDateInterceptor implements Interceptor {
+    private final DateTimeFormatter format = DateTimeFormatter
```

Comment:
My spidey senses are telling me that we should make this static, it's going to get reused over and over - and while the interceptor instances themselves are pretty long-lived, each client will have at least one.

Reply:
Should the interceptors be public API surface?  I'm thinking it's likely an internal thing for right now.  We want to expose as little as possible, and increase the surface area as we need to.